src/components/tiles/text/text-toolbar.tsx

```diff
@@ -53,6 +62,39 @@ export const TextToolbarComponent: React.FC<IProps> = (props: IProps) => {
   const { documentContent, editor, selectedButtons, onIsEnabled, ...others } = props;
   const toolbarSetting = useSettingFromStores("tools", "text") as unknown as string[];
   const enabled = onIsEnabled();
+  const textContent = useContext(TextContentModelContext);
+  const selectedElements = editor?.selectedElements();
+  const variables = getVariables(textContent); 
+  const hasVariable = editor?.isElementActive(kVariableFormat);
+  const selectedVariable = hasVariable ? findSelectedVariable(selectedElements, variables) : undefined;
+  const sharedModel = getOrFindSharedModel(textContent);
+  const highlightedText = (editor && editor.selection) ? Editor.string(editor, editor.selection) : "";
+  
+  const plugins = getAllTextPluginInfos();
+  // Build up a map from plugin => toolbar click handlers.
+  // This assumes the plugin is registering a modal which is not the most generic choice.
+  const pluginModalHandlers: Record<string, ()=> void> = {}; 
```

Comment:
This probably should be more generic. The modalHook is pretty specific for a toolbar button... Also, to keep this somewhat generic, the hook is getting passed all the superset information that the plugins need instead of just what each individual plugin needs. e.g. the `useNewVariableDialog` only needs `addVariable` and `sharedModel` but gets a lot more

Reply:
This will likely also cause the diagram-view module to be pulled into the into the core of CLUE since this file is now importing stuff from `../../../plugins/shared-variables/`.

So this needs some more work so this code can be moved in to `../../../plugins/shared-variables/` somehow. 

@kswenson I can help with this. I'm not sure how this PR will shake out, but I hope the changes would be minor, so they can just be re-applied manually if necessary. So I'll do that as a branch off of this one for the time being.